### PR TITLE
[CI][GCE/5] Add GCE variations of GPU tests

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?

Add GCE variations of GPU tests. Two key things we need to change in order for GPU tests to work:
- Better concurrency control for GPU tests in GCE. GCE has low GPU quota, and between ray start up and auto-scale, jobs competing for resources tend to run into deadlock. With better concurrency control, they can now all run successfully
- The 'dataset_shuffle_push_based_random_shuffle_100tb' test requires a 400TB storage in the cluster. GCE however, currently, has only 200TB. So I change this test to run with 50tb of data in GCE (for now).

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- Testing Strategy
   - https://buildkite.com/ray-project/release-tests-pr/builds/33581 (only one remaining failing tests
   - https://buildkite.com/ray-project/release-tests-pr/builds/33608
   - https://buildkite.com/ray-project/release-tests-pr/builds/33607

### CODE REVIEW

**Comment by can-anyscale** [comment]
Ready to be review, and merge if approved. Thanks!

File: release/air_tests/air_benchmarks/compute_gce_gpu_16.yaml [hunk 1]
```diff
@@ -16,13 +16,10 @@ worker_node_types:
...
-#            Throughput: 1000
-#            VolumeSize: 1000
-#            VolumeType: gp3
+gcp_advanced_configurations_json:
+  instance_properties:
+    disks:
+      - boot: true
+        auto_delete: true
+        initialize_params:
+          disk_size_gb: 800
```
Comment (aslonnie) [inline release/air_tests/air_benchmarks/compute_gce_gpu_16.yaml:25]: nit: eol?

File: release/air_tests/air_benchmarks/compute_gce_gpu_1.yaml [hunk 2]
```diff
@@ -16,12 +16,11 @@ worker_node_types:
...
-#            VolumeSize: 1000
-#            VolumeType: gp3
+gcp_advanced_configurations_json:
+  instance_properties:
+    disks:
+      - boot: true
+        auto_delete: true
+        initialize_params:
+          disk_size_gb: 1000
+
```
Comment (aslonnie) [inline release/air_tests/air_benchmarks/compute_gce_gpu_1.yaml:26]: nit: eol?

**Review by aslonnie — COMMENTED** [review]
test config changes look good. I do not quite understand the `concurrency.py` related changes though.

**Review by krfricke — APPROVED** [review]
Thanks! Looks great. Also like the improvement in the concurrency group logic.

**Pull request merged**